source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -161,11 +163,38 @@ namespace
             {
                 RtValue::fromString(attrType, valueStr, attr.getValue());
             }
+
             readMetadata(elem, PvtObject::ptr<PvtObject>(attr), attrMetadata);
         }
     }
 
-    void createNodeConnections(const vector<NodePtr>& nodeElements, PvtPrim* parent, const PvtRenamingMapper&mapper)
+    void createConnection(PvtOutput* output, PvtInput* input, const string& swizzle, PvtStage* stage)
+    {
+        // Check if a swizzle node should be used in the connection.
+        if (!swizzle.empty())
+        {
```

Comment:
Support for swizzling on connections by inserting a swizzle node.

Since we now validate for matching port data types when making connections we got errors whenever swizzle/channels were used because the data types don't match. This fixes that problem since the swizzle node by design converts the data type as well.

Reply:
This is only for supporting "channels" on _import_. We could add support to _export_ as well, converting any swizzle node to a "channels" attribute. But I think we should wait, since most consumers doens't support the channel attribute. So better to keep any swizzle nodes as explicit nodes for now during file save.